### PR TITLE
Add specific Azure permissions required for Azure SD

### PR DESCRIPTION
**Pull request by jeremiahharbach**
There isn't any documentation that details which specific Azure permissions are required to use Azure SD. I found the proposed permissions to work through trial and error.

### CODE REVIEW

**Comment by beorn7** [comment]
Thank you for your contribution.

Given that the Azure SD is still (and sadly) effectively unmaintained, I'm inclined to just merge this.
@roidelapluie @bboreham @machine424 as the general maintainers of this repo, WDYT?

File: docs/configuration/configuration.md [hunk 1]
```diff
@@ -676,6 +676,13 @@ http_headers:
 
 Azure SD configurations allow retrieving scrape targets from Azure VMs.
 
+The following Azure permissions apply to Azure SD:
```
Comment (machine424) [inline docs/configuration/configuration.md:679]:
```suggestion
The discovery requires at least the following permissions:
```

**Review by machine424 — COMMENTED** [review]
Thanks for this.
(Taking your word for it :))

Does this apply and make sense for all authentication methods (`OAuth, ManagedIdentity, SDK`)?

**Comment by jeremiahharbach** [comment]
> Thanks for this. (Taking your word for it :))
> 
> Does this apply and make sense for all authentication methods (`OAuth, ManagedIdentity, SDK`)?

Yes

**Review by machine424 — APPROVED** [review]
lgtm, thanks again.

**Review by beorn7 — APPROVED** [review]
Test failure is the known problem with a race in the test setup. This is just a doc change, not affected by it, so I'll merge this.

**Pull request merged**